.travis.yml

```diff
@@ -16,14 +16,25 @@ after_success:
   - docker push chengjoanna1729/todo-app:$TRAVIS_COMMIT
 before_deploy:
   - docker push chengjoanna1729/todo-app:latest
+  - wget https://releases.hashicorp.com/terraform/"$TF_VERSION"/terraform_"$TF_VERSION"_linux_amd64.zip
+  - unzip terraform_"$TF_VERSION"_linux_amd64.zip
+  - sudo mv terraform /usr/local/bin/
+  - rm terraform_"$TF_VERSION"_linux_amd64.zip
+  - terraform init
+  - terraform apply -auto-approve
 deploy:
   provider: script
-  script: curl -dH -X POST "$WEBHOOK"
+  script: curl -dH -X POST "$(terraform output -raw webhook)"

```

Comment:
I _think_ this will error because by [default Travis cleans up before a deployment](https://docs.travis-ci.com/user/deployment#uploading-files-and-skip_cleanup), which means the terraform state will be lost. They're [adjusting that in the next version of deployments](https://docs.travis-ci.com/user/deployment-v2) but we should work with what we've got here - we can either skip the cleanup stage, or save the output to an environment variable before reaching the deploy step